### PR TITLE
Handle invalid JSONPath with KeyError, Exceptions, and add test cases

### PR DESCRIPTION
**Pull request by ghost**
Bad `JSONPath` lookups, specifically ones that attempt to access a `list` index against a `dict` value, results in a `KeyError`

```bash
tests/entity/test_entity_service.py::test_get PASSED                                                                                                                                                                                                                                                                                                                                                        [  9%]
tests/entity/test_entity_service.py::test_get_all_by_signal PASSED                                                                                                                                                                                                                                                                                                                                          [ 18%]
tests/entity/test_entity_service.py::test_create PASSED                                                                                                                                                                                                                                                                                                                                                     [ 27%]
tests/entity/test_entity_service.py::test_update PASSED                                                                                                                                                                                                                                                                                                                                                     [ 36%]
tests/entity/test_entity_service.py::test_delete PASSED                                                                                                                                                                                                                                                                                                                                                     [ 45%]
tests/entity/test_entity_service.py::test_find_entities_with_field_and_regex PASSED                                                                                                                                                                                                                                                                                                                         [ 54%]
tests/entity/test_entity_service.py::test_find_entities_with_regex_only PASSED                                                                                                                                                                                                                                                                                                                              [ 63%]
tests/entity/test_entity_service.py::test_find_entities_with_field_only PASSED                                                                                                                                                                                                                                                                                                                              [ 72%]
tests/entity/test_entity_service.py::test_find_entities_with_no_regex_or_field PASSED                                                                                                                                                                                                                                                                                                                       [ 81%]
tests/entity/test_entity_service.py::test_find_entities_handles_key_error PASSED                                                                                                                                                                                                                                                                                                                            [ 90%]
tests/entity/test_entity_service.py::test_find_entities_multiple_entity_types PASSED
```

### CODE REVIEW

**Comment by wssheldon** [comment]
You can call `str` on the `JSONPath` object:

```bash
>>> from jsonpath_ng import jsonpath, parse
>>> jsonpath_expr = parse('foo[*].baz')
>>> str(jsonpath_expr)
'foo.[*].baz'
```

File: src/dispatch/entity/service.py [hunk 1]
```diff
@@ -360,9 +364,18 @@ def _find_entities_by_regex_and_jsonpath_expression(
                                         entity_type=entity_type,
                                         project=signal_instance.project,
                                     )
-                except jsonpath_ng.JSONPathError:
-                    # field not found in signal_instance.raw
-                    pass
+                except KeyError:
+                    log.exception(
```
Comment (kevgliss) [inline src/dispatch/entity/service.py:368]: Should these be logged as exceptions or info? With global entities wouldn't this fire quite a bit, even though it's large expected? Looking for a path that doesn't exist?

Comment (wssheldon) [reply src/dispatch/entity/service.py:368]: I think it's an `exception`. If the key can't found be an empty list is returned.

Comment (wssheldon) [reply src/dispatch/entity/service.py:368]: Eh, I'll downgrade it to `warning`. I do think it's a failure case.

Comment (wssheldon) [reply src/dispatch/entity/service.py]: With the context, should it be `info`, or do you think `warning`?

Comment (kevgliss) [reply src/dispatch/entity/service.py]: I think an exception would indicate something needs to be fixed/handled if we don't expect to take action a warning / info is probably fine imo.

**Pull request merged**